config/prow/cluster/trusted_serviceaccounts.yaml

```diff
@@ -25,6 +25,14 @@ metadata:
 ---
 kind: ServiceAccount
 apiVersion: v1
+metadata:
+  annotations:
+    iam.gke.io/gcp-service-account: gencred-refresher@k8s-prow.iam.gserviceaccount.com
```

Comment:
This GCP SA was created manually, workload identity bound, and required IAM roles were given. The IAM roles given were exactly the same as in https://github.com/kubernetes/test-infra/pull/26293/files#diff-0ff314c33f90e8b1f3e4467d610300b0be4d5f0c39ba22105e6c45c373131302R265